### PR TITLE
closing and required parameters

### PR DESCRIPTION
**Pull request by SamiHiltunen**
Hey,

I propose the following changes:

1. Add a ```Close()``` method for the Recorder. This will allow users to flush the buffers before their application terminates and thus avoid dropping traces.

2. Make ```Project ID```  a required parameter. Recorder throws an error at the moment if WithProjectID("--") is not given which is a bit confusing. Making it a required parameter makes the need for this parameter explicit.

3. Invert the dependency to Google's protobuf client. This allows for easier testing or alternative implementations. Building the client should not probably be the recorder's responsibility.

### CODE REVIEW

File: recorder.go [hunk 1]
```diff
@@ -7,8 +7,8 @@ import (
 	"strconv"
 	"time"
 
-	trace "cloud.google.com/go/trace/apiv1"
 	"github.com/golang/protobuf/ptypes/timestamp"
+	gax "github.com/googleapis/gax-go"
```
Comment (heltonmarx) [inline recorder.go]: One concern about `gax` is : **This project is currently experimental and not supported**

Comment (SamiHiltunen) [inline recorder.go]: It's a dependency of the protobuf client we are using so, not sure we can do much there.

Comment (heltonmarx) [inline recorder.go]: ok !

File: README.md [hunk 2]
```diff
@@ -23,16 +23,28 @@ First of all, you need to init Global Tracer with GCloud Tracer:
...
+    client, err := trace.NewClient(context.Background() /*auth options here if necessary*/)
+    if err != nil {
+      log.Fatalf("error creating a tracing client: %v", err)
+    }
+
+    recorder, err := gcloudtracer.NewRecorder(context.Background(), "gcp-project-id", client)
+    if err != nil {
+      log.Fatalf("error creating a recorder: %v", err)
+    }
+
```
Comment (heltonmarx) [inline README.md]: add `defer recorder.Close()` here

**Pull request merged**